pkg/scaffold/v2/dockerfile.go

```diff
@@ -60,5 +68,12 @@ RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager
 FROM gcr.io/distroless/static:latest
 WORKDIR /
 COPY --from=builder /workspace/manager .
-ENTRYPOINT ["/manager"]
-`
+COPY build/bin /usr/local/bin
+
+ENV OPERATOR=/workspace/manager \
+    USER_UID=1001 \
+    USER_NAME=manager
+
+RUN  /usr/local/bin/user_setup
+ENTRYPOINT ["/usr/local/bin/entrypoint"]
```

Comment:
What's up with the rest of these changes?

Reply:
Hi @DirectXMan12, 

Really tks for your time. But it still in WIP  so I did not think that you would review it. Few changes were made, please feel free to check them. However, shows that I am still with some problem to exec the /manager in the Entrypoint.

Reply:
Just some background on this:

By default in OpenShift, containers are run as uid=\<random UID\>, gid=0

The user setup script sets up unix perms so that gid=0 processes can access the user's home directory and update `/etc/passwd`.

When the container starts, the custom entrypoint adds an entry to `/etc/passwd` if the user is not already present, which will typically be the case since the user setup script doesn't actually create the user.

The outcomes of these are:
1. In non-OpenShift clusters, the operator is executed as `manager:root` (`1001:0`)
2. In OpenShift clusters the operator is executed as `manager:root` (`<random uid>:0`)

In both cases, the operator files and executables are owned by `1001:0` and have permissions that allow gid=0 to read/write/execute as necessary, thus allowing random uids to run correctly as long as they're in the `root` group.

Reply:
This is how it currently works in SDK, but we may be able to simplify this significantly.

Reply:
ah, yeah.

So, distroless/static has a "nonroot" user that already is set up with a passwd entry, etc.  We can make use of that.  Since we're in static, we don't have (or want) things like a shell.

Reply:
Really tks. The changes suggested are done and all shows working fine in the OCO without the shell as well. 

c/c @joelanford @DirectXMan12

Reply:
closing it since shows all sorted.